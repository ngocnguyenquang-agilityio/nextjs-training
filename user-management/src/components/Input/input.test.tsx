import { fireEvent, render, screen } from '@testing-library/react';

// Components
import { Input } from '.';

describe('Button Component', () => {
  test('Should render successfully', () => {
    const comp = render(<Input />);
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    const comp = render(<Input />);
    expect(comp).toMatchSnapshot();
  });

  test('Should render error input', () => {
    render(<Input type="text" error />);
    const input = screen.getByRole('textbox');
    expect(input.getAttribute('class')).toMatch(
      /outline outline-1 outline-offset-1 outline-red-600 focus:outline-2 focus:outline-red-600/gi
    );
  });

  test('Should able to type', () => {
    const { getByPlaceholderText } = render(<Input type="text" placeholder="Type something" />);
    const input = getByPlaceholderText('Type something') as HTMLInputElement;
    const value = 'typing value';

    fireEvent.change(input, { target: { value: value } });
    expect(input.value).toBe(value);
  });

  test('Should call onChange function', () => {
    const mockOnChange = jest.fn();
    render(<Input type="text" onChange={mockOnChange} />);
    const input = screen.getByRole('textbox');
    fireEvent.change(input, { target: { value: 'Typing value' } });
    expect(mockOnChange.mock.calls.length).toBe(1);
  });
});
