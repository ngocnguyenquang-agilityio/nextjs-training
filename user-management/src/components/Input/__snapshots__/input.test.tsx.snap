// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="bg-transparent block w-full border text-black text-sm font-md rounded-lg p-2.5 border-gray-500 focus:ring-blue-500 outline-none focus:border-blue-500 placeholder-gray-400 focus:outline-2 focus:outline-blue-500 focus:border-none false undefined"
      />
    </div>
  </body>,
  "container": <div>
    <input
      class="bg-transparent block w-full border text-black text-sm font-md rounded-lg p-2.5 border-gray-500 focus:ring-blue-500 outline-none focus:border-blue-500 placeholder-gray-400 focus:outline-2 focus:outline-blue-500 focus:border-none false undefined"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
